opm/simulators/flow/ConvergenceOutputConfiguration.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+  Copyright 2022 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/simulators/flow/ConvergenceOutputConfiguration.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <regex>
+#include <stdexcept>
+#include <string_view>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <fmt/format.h>
+
+namespace {
+    std::vector<std::string> tokenizeOptionValues(std::string_view options)
+    {
+        const auto split = std::regex { R"(\s*,\s*)" };
+        return {
+            std::cregex_token_iterator {
+                options.begin(), options.end(), split, -1
+            },
+            std::cregex_token_iterator {}
+        };
+    }
+
+    [[noreturn]] void reportUnsupportedOptionValuesAndThrow(std::vector<std::string> unsupp)
+    {
+        std::sort(unsupp.begin(), unsupp.end());
+        auto u = std::unique(unsupp.begin(), unsupp.end());
+
+        const auto pl = (unsupp.size() != 1) ? "s" : "";
+
+        throw std::invalid_argument {
+            fmt::format("Unsupported convergence output "
+                        "option value{}: {}\n"
+                        "Supported values are \"none\", "
+                        "\"steps\", and \"iterations\"", pl,
+                        fmt::join(unsupp.begin(), u, ", "))
```

Comment:
I think we should also print what command line parameter we are talking about. Otherwise users are lost.

Reply:
> I think we should also print what command line parameter we are talking about. Otherwise users are lost.

I've pushed an update to that effect.